serde/de.py

```diff
@@ -274,7 +274,7 @@ def wrap(cls: type[T]) -> type[T]:
         for typ in iter_types(cls):
             # When we encounter a dataclass not marked with deserialize, then also generate
             # deserialize functions for it.
-            if is_dataclass_without_de(typ):
+            if is_dataclass_without_de(typ) and typ is not cls:
```

Comment:
what was happening is, `@deserialize` was called twice. The second call here does not pass `type_check` in, so it became `type_check=strict`

I wonder we should pass all the class attributes from parent class :thinking: